### PR TITLE
Ported xen-api-sdk to dune and fixed warnings.

### PR DESCRIPTION
**Pull request by kc284**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -41,22 +41,22 @@ ifndef SR_XML
 endif
 
 build:
-	jbuilder build --dev \
+	dune build --dev \
```
Comment (mseri) [inline Makefile]: —dev is the default in dune. You need to add —release to the other dune build invocations

**Comment by lindig** [comment]
I don't understand the Travis failure

```
[ERROR] The sources of the following couldn't be obtained, aborting:
          - xapi-stdext-monadic.1.0.0
```

From what I can the corresponding Opam file points to the right tarball:

```

**Comment by mseri** [comment]
It is usually a network issue when you see that error. On osx can be an issue with an old openssl and opam if you are using opam = 2.0.0

**Comment by lindig** [comment]
Travis now passed - @mseri was right.

**Comment by kc284** [comment]
@lindig regarding the alignment, the above changes have been ocp-indented.

**Comment by lindig** [comment]
ocp-indent does not affect the right hand-side; see here for example:
https://github.com/xapi-project/xen-api-sdk/pull/154/files#diff-49b380ccf20e5df18f6545257b853e4aR1516

**Comment by kc284** [comment]
Aaah, sorry, I read the comment quickly and the *right-hand* bit didn;t register with me. Ok, it's not important, it can be rectified in future alongside other code changes.

**Comment by edwintorok** [comment]
another option would be ocamlformat, which reformats the whole AST, but the changes can be quite big and not limited to just the lines you changed.

**Pull request merged**